plugins/filter/filters.py

```diff
@@ -214,6 +214,9 @@ def listener_properties(self, listeners_dict, default_ssl_enabled,
                     'io.confluent.kafka.server.plugins.auth.token.TokenBearerServerLoginCallbackHandler'
                 final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] =\
                     'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required publicKeyPath=\"' + oauth_pem_path + '\";'
+                
+            if self.normalize_sasl_protocol(listeners_dict[listener].get('sasl_protocol', default_sasl_protocol)) == 'OAUTHBEARER' and sso != 'none':    
```

Comment:
This principal builder would work without SSO as well.

If you simply merge this setting to the above `if` condition, that should alos work fine. 

Keeping it separate is also good.

Reply:
how is this Principal builder used without SSO? Is it simply ignored when we don't have SSO enabled or it is used in MDS?

Reply:
your above comment makes this principal builder independent of SSO, so Im trying to understand how it would affect our current RBAC implementation

Reply:
This principal builder uses the default principal builder as delegate. And it is exactly similar to default principal builder with added support of groups field in bearer token.

Even for bearer token without groups claim, the build() call would be delegated to default kafka principal builder.

Reply:
Just wanted to add that it's supported in only version 7.5 onwards and we didn't merge it in older versions of Kafka.

Reply:
You summed it rightly.